sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -681,4 +763,168 @@ String getParentLink() {
     String getLink() {
         return this.link;
     }
+
+    private Mono<CosmosAsyncItemResponse<Object>> deleteItem(
+        String itemId,
+        RequestOptions requestOptions,
+        Context context) {
+        String spanName = "deleteItem." + this.getId();
+        Map<String, String> tracingAttributes = new HashMap<String, String>() {{
+            put(TracerProvider.DB_TYPE, TracerProvider.DB_TYPE_VALUE);
+            put(TracerProvider.DB_INSTANCE, getId());
+            put(TracerProvider.DB_URL, database.getClient().getServiceEndpoint());
+            put(TracerProvider.DB_STATEMENT, spanName);
+        }};
+
+        Mono<CosmosAsyncItemResponse<Object>> responseMono =  this.getDatabase()
+            .getDocClientWrapper()
+            .deleteDocument(getItemLink(itemId), requestOptions)
+            .map(response -> ModelBridgeInternal.createCosmosAsyncItemResponseWithObjectType(response))
+            .single();
+        return database.getClient().getTracerProvider().traceEnabledCosmosItemResponsePublisher(responseMono,
+            tracingAttributes,
+            context, spanName);
+    }
+
+    private <T> Mono<CosmosAsyncItemResponse<T>> replaceItem(
+        Class<T> itemType,
+        String itemId,
+        Document doc,
+        CosmosItemRequestOptions options,
+        Context context) {
+        String spanName = "replaceItem." + this.getId();
+        Map<String, String> tracingAttributes = new HashMap<String, String>() {{
+            put(TracerProvider.DB_TYPE, TracerProvider.DB_TYPE_VALUE);
+            put(TracerProvider.DB_INSTANCE, getId());
+            put(TracerProvider.DB_URL, database.getClient().getServiceEndpoint());
+            put(TracerProvider.DB_STATEMENT, spanName);
+        }};
```

Comment:
exactly same code as above, can we factor out?

```java
Map<String, String> createTracingMap(...) {
   ...
}

and then invoke createTracingMap() here
```

Reply:
Done